Prettify documentation

gromakovsky opened the pull request:
## Description

Problem: there are some typos and subjective minor issues in README.
Solution: fix them as I see fit.



<!--
Describes the nature of your changes. If they are substantial, you should
further subdivide this into a section describing the problem you are solving and
another describing your solution.
-->

## Related issue(s)

<!--
- Short description of how the PR relates to the issue, including an issue link.
For example
- Fixed #100500 by adding lenses to exported items

Write 'None' if there are no related issues (which is discouraged).
-->

I am doing it for https://issues.serokell.io/issue/EDNA-137

## :white_check_mark: Checklist for your Pull Request

<!--
Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

If you don't set a checkmark (e. g. don't add a test for new functionality),
you must be able to justify that.
-->

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

gromakovsky committed:
Prettify README

Problem: there are some typos and subjective minor issues in README.
Solution: fix them as I see fit.

gromakovsky committed:
Provide more documentation about configuration

Problem: there is almost no documentation about configuration
of `edna-server` and the whole app. One may run `edna-server --help`
but it's not so trivial if you use Docker images. Besides,
Docker-based approach has some other means of configuration.

Solution: add a subsection about that to `deployment/README.md`.

gromakovsky committed:
Remove quotes from `.env` file

Problem: docker-compose seems to pass values in quotes in env-file
in such a way that the underlying application sees this quotes,
as if they were escaped. Apparently it doesn't affect our case because
we explicitly forward all variables using `environment`, but some
changes to the configuration may cause quotes to be passed to
`edna-server` and cause incorrect behavior (e. g. if there are quotes
around MDe links).

Solution: remove all quotes from `.env` file, it doesn't seem to spoil
or break anything, but seems safer this way. Add a comment asking
to be careful with quotes.

balsoft committed:
Add information about building with Nix

gromakovsky commented on default.nix:
```diff
@@ -2,7 +2,9 @@
+    "https://github.com/edolstra/flake-compat/archive/master.tar.gz") {
+      src = builtins.fetchGit ./.;
+    }).defaultNix;
+in flake // flake.packages.${builtins.currentSystem}
```
AFAIU the gist of this change is `// flake.packages.${builtins.currentSystem}`. Why is it needed? How is it related to "Add information about building with Nix"?

balsoft replied:
It makes using pre-flake Nix easier. I really don't want to have `nix-build -A packages.x86_64-linux.backend-server` instead of `nix-build -A backend-server` in the readme :)

To reiterate, it re-exports all packages for the current platform as top-level attributes.

gromakovsky replied:
Ah, ok, thanks.

gromakovsky committed:
Change copyright to copyleft in the footer

Problem: our code is licensed on terms of AGPL-3.0 which is a copyleft
license (https://en.wikipedia.org/wiki/Copyleft) and it's common
to use the copyleft instead of the copyright sign in such case.
This way it's clearer that people can distribute and modify
this intellectual property.
However, we ahve the copyright sign in the footer.

Solution: use CSS to rotate the copyright sign. While there is the
copyleft symbol in Unicode, we are not sure how widely it's supported,
so using CSS seems safer (and admitely more tiresome).

worm2fed commented on README.md:
```diff
@@ -12,24 +12,42 @@
 * [backend](/backend/) is a web server implemented in Haskell.
-* [frontend](/frontend/) is web UI implemented in TypeScript.
-* [analysis](/analysis/) contains Python code to analyse data submitted to Edna.
+* [frontend](/frontend/) is a web UI implemented in TypeScript.
```
I think it'll be better to use `a web app` or `web client app` here..

worm2fed commented on README.md:
```diff
@@ -12,24 +12,42 @@
 * [frontend](https://github.com/orgs/serokell/packages/container/package/edna-frontend)
 
-Note that some additional configuration is needed: you need to launch PostgreSQL DB server, forward ports, provide a config file.
+Note that some additional configuration is needed: you need to launch PostgreSQL DB server, forward ports and provide appropriate configuration via a config file or environment variables.
```
or command line arguments)

worm2fed commented on deployment/README.md:
```diff
@@ -22,4 +22,21 @@ By default `docker-compose` will pull Edna images from GitHub Container Registry
+  + `EDNA_*` options match their counterparts from the configuration file.
+
+These are the most significant configuration options you may want to change:
+* `EDNA_API_LISTEN_ADDR` is the network address where the backend server will be bound.
```
If somebody change it, there are can be a problem with frontend app which listens barcoded addr and port.. maybe we have to allow frontend app read some env variables to set listen addr and port

gromakovsky replied:
I want to merge it before today's demo (i. e. ASAP), so I'll just remove it from this list for now.

worm2fed replied:
@gromakovsky okay
but create an issue please, in case it will be forgotten

gromakovsky replied:
https://issues.serokell.io/issue/EDNA-146

worm2fed commented on frontend/README.md:
```diff
@@ -10,16 +10,16 @@
 
 `yarn serve` runs the dev server.
-To operate correctly, it's needed to have backend launched on `8080` port.
+You need to launch backend on `8080` port for the server to operate correctly.
```
we can change backend port but cannot tell to frontend which port to use

gromakovsky replied:
Oh, it looks wrong, I think the port should be 9000, it's specified in `webpack.dev.js`, let me fix it.
